internal/config/substitutions.go

```diff
@@ -45,39 +45,67 @@ func SubLookupFileData(configs *[]load.Config, config load.Config) error {
 
 	// create a new config file per
 	for _, item := range jsonOut {
-		switch obj := item.(type) {
-		case map[string]interface{}:
-			tmpCfgStr := string(tmpCfgBytes)
-			variableReplaces := regexp.MustCompile(`\${lf:.*?}`).FindAllString(tmpCfgStr, -1)
-			replaceOccurred := false
-			for _, variableReplace := range variableReplaces {
-				variableKey := strings.TrimSuffix(strings.Split(variableReplace, "${lf:")[1], "}") // eg. "channel"
-				if obj[variableKey] != nil {
-					tmpCfgStr = strings.Replace(tmpCfgStr, variableReplace, fmt.Sprintf("%v", obj[variableKey]), -1)
-					replaceOccurred = true
-				}
+		tmpCfgStr := string(tmpCfgBytes)
+		newCfg, err := fillTemplateConfigWithValues(item, tmpCfgStr)
+
+		if err != nil {
+			load.Logrus.WithFields(logrus.Fields{
+				"file":       config.LookupFile,
+				"name":       config.Name,
+				"suggestion": "check for errors or run yaml lint against the below output",
+			}).WithError(err).Error("config: new lookup file unmarshal failed")
+			load.Logrus.Error(tmpCfgStr)
+		}
+
+		if newCfg != nil {
+			*configs = append(*configs, *newCfg)
+		}
+	}
+	return nil
+}
+
+func fillTemplateConfigWithValues(values interface{}, configTemplate string) (*load.Config, error) {
+	switch obj := values.(type) {
+	case map[string]interface{}:
+		variableReplaces := regexp.MustCompile(`\${lf:.*?}`).FindAllString(configTemplate, -1)
+		replaceOccurred := false
+		for _, variableReplace := range variableReplaces {
+			variableKey := strings.TrimSuffix(strings.Split(variableReplace, "${lf:")[1], "}") // eg. "channel"
+			if obj[variableKey] != nil {
+				value := obj[variableKey]
+				configTemplate = strings.Replace(
+					configTemplate,
+					variableReplace,
+					fmt.Sprintf(getSprintfTemplateForVariable(value), value),
+					-1)
+				replaceOccurred = true
 			}
-			// if replace occurred convert string to config yaml and reload
-			if replaceOccurred {
-				newCfg, err := ReadYML(tmpCfgStr)
-				if err != nil {
-
-					load.Logrus.WithFields(logrus.Fields{
-						"file":       config.LookupFile,
-						"name":       config.Name,
-						"suggestion": "check for errors or run yaml lint against the below output",
-					}).WithError(err).Error("config: new lookup file unmarshal failed")
-					load.Logrus.Error(tmpCfgStr)
-
-				} else {
-					*configs = append(*configs, newCfg)
-				}
+		}
+		// if replace occurred convert string to values yaml and reload
+		if replaceOccurred {
+			newCfg, err := ReadYML(configTemplate)
+			if err != nil {
+				return nil, err
 			}
-		default:
-			load.Logrus.Debug("config: lookup file needs to contain an array of objects")
+			return &newCfg, nil
 		}
+	default:
+		load.Logrus.Debug("config: lookup file needs to contain an array of objects")
+	}
+	return nil, nil
+}
+
+func getSprintfTemplateForVariable(value interface{}) string {
```

Comment:
there's no need to have such a specific name for the function. maybe a simple "toString(..)" or "formatToString(...)".
also perhaps instead of returning the format string, why not return the formatted value already?